src/libraries/System.ComponentModel.Composition/src/System/ComponentModel/Composition/ReflectionModel/ImportType.cs

```diff
@@ -172,14 +172,15 @@ private static bool IsLazyGenericType(Type genericType)
 
         private static bool TryGetCastFunction(Type genericType, bool isOpenGeneric, Type[] arguments, [NotNullWhen(true)] out Func<Export, object>? castFunction)
         {
-            castFunction = null;
+            castFunction = null!; // TODO2
```

Comment:
:memo: the `[NotNullWhen(...)]` annotation on method seems incorrect

Reply:
Yes, please remove it.  The dangers of the compiler not having checked those annotations ;)